rust/connlib/clients/shared/src/control.rs

```diff
@@ -229,8 +229,10 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
                 };
 
                 tokio::spawn(async move {
-                    if let Err(e) = upload(path, url).await {
+                    if let Err(e) = upload(path.clone(), url).await {
                         tracing::warn!("Failed to upload log file: {e}")
+                    } else if let Err(e) = tokio::fs::remove_file(&path).await {
+                        tracing::warn!("Failed to remove log file: {e}")
                     }
```

Comment:
Prefer early return over chained ifs :)

```suggestion
                        tracing::warn!("Failed to upload log file: {e}");
                        return;
                    }
                    
                    if let Err(e) = tokio::fs::remove_file(&path).await {
                        tracing::warn!("Failed to remove log file: {e}")
                    }
```

Reply:
Just to be clear, we will never attempt to upload the file again (with the current logic) but I think it makes sense to keep it around.

Reply:
Yeah, same

Reply:
Updated